tools/swift-refactor/swift-refactor.cpp

```diff
@@ -74,6 +74,7 @@ Action(llvm::cl::desc("kind:"), llvm::cl::init(RefactoringKind::None),
                       "replace-bodies-with-fatalError", "Perform trailing closure refactoring"),
            clEnumValN(RefactoringKind::MemberwiseInitLocalRefactoring, "memberwise-init", "Generate member wise initializer"),
            clEnumValN(RefactoringKind::AddEquatableConformance, "add-equatable-conformance", "Add Equatable conformance"),
+           clEnumValN(RefactoringKind::AddExplicitCodableImplementation, "add-explicit-codable-implementation", "Add Explicit Codable Implementation"),
```

Comment:
Note: this is behaves different from the existing `AddEquatableConformance` action. That is, for Codable we really want the synthesized implementation. This action doesn't make your type conform to `Codable` (which is easy, just make it conform to it manually). It's geared towards "it conforms to Codable, and now I want to insert what the compiler generated".